internal/kms/vault_sa.go

```diff
@@ -310,7 +314,7 @@ func (kms *vaultTenantSA) getToken() (string, error) {
 		}
 	}
 
-	return kms.createToken(sa, c)
+	return "", fmt.Errorf("failed to find token in ServiceAccount %s/%s", kms.Tenant, kms.tenantSAName)
```

Comment:
```suggestion
	err = fmt.Errorf("failed to find token in ServiceAccount %s/%s", kms.Tenant, kms.tenantSAName)
	if createTokenErr != nil {
		err = fmt.Errorf(%w and failed to create token: %w",err, createTokenErr)
	}
	return "", err
```

Reply:
We only reach here if createTokenErr is not `nil`. I felt `if` here is redundant. Included it as part of final error.